[blog] Blog post for MUI X v6 alpha zero (#34424)
mui/material-ui: docs/pages/blog/mui-x-v6-alpha-zero.md

  - [Use the Data Grid internal components outside the grid](https://github.com/mui/mui-x/issues/2522).

- **Improved look & feel**  
   We’re polishing edges in terms of design and usability.

  - [New column menu](https://github.com/mui/mui-x/issues/4929).
  - [New column visibility panel](https://github.com/mui/mui-x/issues/5700).
  - [Column resize bar only on hover](https://github.com/mui/mui-x/issues/1623).

- **Rockstar feature**
  - [Copy and paste from/to multiple cells](https://github.com/mui/mui-x/issues/199).

### Date and Time Pickers

- **Improved UX**  
   In the previous version we focused on developer experience and overall stability of the components. Now we’re tackling ways to improve usability.

  - [Remove the clock view on time pickers (for desktop)](https://github.com/mui/mui-x/issues/4483).
  - [Visually edit a range by dragging date markers](https://github.com/mui/mui-x/issues/5311).
  - [Range shortcuts](https://github.com/mui/mui-x/issues/4563).
  - [A new text input for date and time values, retiring the mask solution (early preview)](https://next.mui.com/x/react-date-pickers/date-field/).

> - **Improved Customizabilitys**  

Reviewer: I'm sorry if I did not follow the history on this one, but a nitpick: Is `Customizabilitys` an [actual word](https://en.wiktionary.org/wiki/customizability)? 🤔 
I'd suggest going with `Improved Customization` instead. 🙈

> - **Improved Customizabilitys**  

Author: ```suggestion
- **Improved Customization**  
```

---
title: MUI X v6.0.0-alpha.0

> description: We’re kicking off the development of the next version.

Reviewer 2: How could we describe this announcement in the title and description to make it more exciting? Why should I, as an MUI X user and casual reader of the blog, be excited about this alpha zero? What's in it for me?

Author: There's the issue that the alpha.0 package per se, doesn't bring anything new.
But I guess we can use v6 in general for this.

```suggestion
title: MUI X v6 Alpha
description: Improved customization and usability. We’re kicking off the next major update for the advanced components.
```

Author: @samuelsycamore, I'm commiting this one, feel free to suggest anything else.

date: 2022-09-30T00:00:00.000Z
authors: ['josefreitas']

> tags: ['News']

Reviewer 3: ```suggestion
tags: ['MUI X', 'News']
```

---

> We’re kicking off the development of [MUI X v6](https://github.com/mui/mui-x/releases/tag/v6.0.0-alpha.0). Both the Data Grid and Date Pickers are getting a new major version. We want to share our plans and invite you to join us on this next step.

Reviewer 2: ```suggestion
We’re kicking off the development of [MUI X v6](https://github.com/mui/mui-x/releases/tag/v6.0.0-alpha.0). Both the Data Grid and the Date and Time Pickers are getting a new major version. We want to share our plans and invite you to join us on this next step.
```

:::warning

> 

Reviewer 4: ```suggestion
```

> It’s important to notice that only MUI X is getting a new version; The core components remain in v5 for a while longer.

Reviewer 2: Cleaning up the grammar and being more explicit w/r/t "core components."

```suggestion
It’s important to note that only MUI X is getting a new version—MUI Core (including Material UI) will remain in v5 for now.
```

You can rest assured that all MUI X v6 components will be fully compatible with MUI Core v5.
::::

> ## How is it going to work?

Reviewer 2: To be more clear.

```suggestion
## What's the plan to get to a stable release?
```

> Before officially releasing v6, we'll go through a few months with pre-releases. We’ll follow the current weekly release process and [semver versioning](https://semver.org/); The difference now is that instead of v5, we’ll ship v6 pre-release packages.

Reviewer 3: We only use straight quotes for consistency

```suggestion
Before officially releasing v6, we'll go through a few months with pre-releases. We'll follow the current weekly release process and [semver versioning](https://semver.org/); The difference now is that instead of v5, we'll ship v6 pre-release packages.
```

There should be a way to add a rule with markdownlint about it.

> Before officially releasing v6, we'll go through a few months with pre-releases. We'll follow the current weekly release process and [semver versioning](https://semver.org/); The difference now is that instead of v5, we'll ship v6 pre-release packages.

Reviewer 2: No need to capitalize after a semi-colon.

```suggestion
Before officially releasing v6, we'll go through a few months with pre-releases. We'll follow the current weekly release process and [semver versioning](https://semver.org/); the difference now is that instead of v5, we'll ship v6 pre-release packages.
```

First, in the alpha phase, we’ll introduce all the breaking changes planned for this major. We’ll be exploring a bit, so it’s expected that some APIs will be unstable, not in terms of functionality, but we may need to rename or adjust parameters. This phase is planned to take about two months.

> Next, comes the beta phase, where the APIs are more stable, and we focus on bug fixing and polishing the hard edges. This phase is planned to take about one month.

Reviewer 2: Maintain future tense.

```suggestion
Next comes the beta phase, where the APIs will be more stable, and we'll focus on fixing bugs and polishing the hard edges. This phase is planned to take about one month.
```

:::info
Disclaimer: We're working with 2 and 1 months for the alpha and beta phase respectively, but that's just a **reference**.

> We operate in a dynamic environment that's subject to change. We may need to do more iterations or change the referenced time windows to deliver the great product we want.

Reviewer 2: ```suggestion
We operate in a dynamic environment that's subject to change, and the actual timeline could vary.
We may need to do more iterations in the interest of delivering the best possible product.
```

:::

> At last, we’ll release the first `v6.0.0` stable, and we’ll continue improving the components and adding features, but without making any new breaking changes until the next major version.

Reviewer 2: To be more explicit.

```suggestion
Finally, we’ll release the first `v6.0.0` stable, and we’ll continue improving the components and adding features, but without making any new breaking changes until the next major version.
We hope to reach the stable release by the end of 2022.
```

> We’re establishing a one-year cycle for major versions, meaning the next major comes about one year after this release.

Reviewer 2: ```suggestion
We’re establishing a one-year cycle for major versions, meaning the next major will arrive about one year after this release.
```

## What happens to v5?

> During v6 pre-releases, v5 is the official current major, and it will remain supported during the next version development.

Reviewer 2: ```suggestion
During v6 pre-releases, v5 will continue to be the official current major, and it will remain supported during this time.
```

> However, we’ll release new v5 versions by demand, only to patch bugs, or add community contributions.

Reviewer 2: ```suggestion
However, we will only release new v5 versions as needed to patch bugs or add community contributions.
```

> All new features and enhacements go to v6, and after its first stable release, v5 is officially in [Long-term support](https://mui.com/versions/#long-term-support-lts).

Reviewer 2: ```suggestion
All new features and enhancements will go to v6, and after its first stable release, v5 will officially transition to [long-term support](https://mui.com/versions/#long-term-support-lts) status.
```

## Where's the v6 documentation?

As mentioned above, v5 is still the official version, so by default, the documentation shows v5 features and API.
The next version's documentation is in the `next` subdomain.

- [https://next.mui.com/x/react-data-grid/](https://next.mui.com/x/react-data-grid/)
- [https://next.mui.com/x/react-date-pickers/](https://next.mui.com/x/react-date-pickers/)

## What’s coming next?

> The following is a list of enhancements on the pipeline for the next few months. It contains the highlights that will make into the first versions.

Reviewer 2: Fixing "in the pipeline" grammar.

```suggestion
The following is a list of enhancements in the pipeline for v6. It contains the highlights that will be included in the first versions.
```

### Data Grid

- **Extended customization abilities**.  

>    We want to empower more users to suit the data grid to complex use cases.

Reviewer 2: ```suggestion
   We want to empower more users to employ the Data Grid for complex use cases.
```

  - [ApiRef in the community package](https://github.com/mui/mui-x/issues/6147).
  - [Filtering on header](https://github.com/mui/mui-x/issues/6247).
  - [Use the Data Grid internal components outside the grid](https://github.com/mui/mui-x/issues/2522).

> - **Improved Design**  

Reviewer 3: Design could be confused with design in the general sens, like design architecture

```suggestion
- **Improved look & feel**  
```

Reviewer 3: But maybe it's ok
```suggestion
- **Improved design**  
```

   We’re polishing edges in terms of design and usability.

  - [New column menu](https://github.com/mui/mui-x/issues/4929).
  - [New column visibility panel](https://github.com/mui/mui-x/issues/5700).
  - [Column Resize Bar only on hover](https://github.com/mui/mui-x/issues/1623).

- **Rockstar feature**
  - [Copy and paste from/to multiple cells](https://github.com/mui/mui-x/issues/199).

> ### Date pickers

Reviewer 3: ```suggestion
### Date Pickers
```

> ### Date Pickers

Reviewer 2: ```suggestion
### Date and Time Pickers
```

- **Improved UX**  

>    On the previous version we focused a lot on developer experience and overall stability of the components. Now we’re pushing the lines to improve usability.

Reviewer 2: ```suggestion
   In the previous version we focused on developer experience and overall stability of the components. Now we’re tackling ways to improve usability.
```

  - [Remove the clock view on time pickers (for desktop)](https://github.com/mui/mui-x/issues/4483).
  - [Visually edit a range by dragging date markers](https://github.com/mui/mui-x/issues/5311).
  - [Range shortcuts](https://github.com/mui/mui-x/issues/4563).
  - [A new text input for date and time values, retiring the mask solution (early preview)](https://next.mui.com/x/react-date-pickers/date-field/).

:::info
You can check our [roadmap](https://github.com/mui/mui-x/projects/1) for the full live list.
:::

## How to migrate?

> We’ve prepared a [migration guide](https://mui.com/x/react-data-grid/migration-v5/), and we’ll feed the page over time, as we make any breaking changes during the pre-releases. It lists every update you need to do on your code to use the most recent packages.

Reviewer 2: ```suggestion
We’ve prepared a [migration guide](https://mui.com/x/react-data-grid/migration-v5/), and we’ll continuously update it as we make any breaking changes during the pre-releases. It lists every update you need to make to your code to use the most recent packages.
```

We highly encourage you to try the new version. It is, after all, an improvement over the solid foundation we established with v5.

## How to get involved?

> Please consider joining our alpha testers group channel and [giving us](https://forms.gle/vsBv6CLPz9h57xg8A) an user interview.  

Reviewer 2: ```suggestion
Please consider joining our alpha testers group channel and [connecting with us](https://forms.gle/vsBv6CLPz9h57xg8A) for a user interview.  
```

> You'll get an insider's perspective of the development and can help us iterate early in the new features.

Reviewer 2: ```suggestion
You'll get an insider's perspective on the development, and you'll be able to help us iterate early on the new features.
```

> As usual, we're glad to listen all the feedback, and you can also participate in the discussion by commenting on the features or reporting bugs in our [GitHub repository](https://github.com/mui/mui-x/issues/new/choose). Finally, you can follow every step of the development through our [changelog](https://github.com/mui/mui-x/releases).

Reviewer 3: I guess we are missing the actual changelog of v6.0.0-alpha.0? Either as a reference or inclined.

Author: I added a link to the alpha zero release on first section of the blog.

> As usual, we're glad to listen all the feedback, and you can also participate in the discussion by commenting on the features or reporting bugs in our [GitHub repository](https://github.com/mui/mui-x/issues/new/choose). Finally, you can follow every step of the development through our [changelog](https://github.com/mui/mui-x/releases).

Reviewer 2: ```suggestion
As always, we're happy to get your feedback. You can participate in the discussion by commenting on new features or reporting bugs in our [GitHub repository](https://github.com/mui/mui-x/issues/new/choose). Finally, you can follow every step of the development through our [changelog](https://github.com/mui/mui-x/releases).
```

---

> title: A major update is coming for the advanced components.

Reviewer 2: Sorry I dropped the ball on Friday! I like the direction you're going in with this title. What if we added a hook to address the reader and hint at our call to action?

```suggestion
title: A major update is coming for MUI X—and you can get involved
```

> description: Improved customization and usability. We’re kicking off the development of MUI X v6.

Reviewer 2: Reframing this to center the reader and underscore the CTA once again.

```suggestion
description: Let us know what you want to see in MUI X v6 as we begin the alpha phase of development.
```

date: 2022-09-30T00:00:00.000Z
authors: ['josefreitas']
tags: ['MUI X', 'News']
---

> We’re kicking off the development of [MUI X v6](https://github.com/mui/mui-x/releases/tag/v6.0.0-alpha.0). Both the Data Grid and the Date and Time Pickers are getting a new major version. We want to share our plans and invite you to join us on this next step.

Reviewer 3: I assume it's really a new major once stable?

```suggestion
We’re kicking off the development of [MUI X v6](https://github.com/mui/mui-x/releases/tag/v6.0.0-alpha.0). Both the Data Grid and the Date and Time Pickers will get a new major version. We want to share our plans and invite you to join us on this next step.
```

:::warning

> It’s important to note that only MUI X is getting a new version—MUI Core (including Material UI) will remain in v5 for now.

Reviewer 3: Non-breaking space?

```suggestion
It’s important to note that only MUI X is getting a new version—MUI Core (including Material UI) will remain in v5 for now.
```

**Before**
<img width="733" alt="Screenshot 2022-10-06 at 20 59 34" src="https://user-images.githubusercontent.com/3165635/194396517-99981734-1bae-4086-8e84-7914e9ca4332.png">

**After**
<img width="714" alt="Screenshot 2022-10-06 at 21 00 13" src="https://user-images.githubusercontent.com/3165635/194396611-c7e919b2-22af-4752-8eeb-b4da07233a52.png">

## Where's the v6 documentation?

As mentioned above, v5 is still the official version, so by default, the documentation shows v5 features and API.
The next version's documentation is in the `next` subdomain.

- [https://next.mui.com/x/react-data-grid/](https://next.mui.com/x/react-data-grid/)
- [https://next.mui.com/x/react-date-pickers/](https://next.mui.com/x/react-date-pickers/)

## What’s coming next?

The following is a list of enhancements in the pipeline for v6. It contains the highlights that will be included in the first versions.

### Data Grid

- **Extended customization abilities**.  
   We want to empower more users to employ the Data Grid for complex use cases.

  - [ApiRef in the community package](https://github.com/mui/mui-x/issues/6147).
  - [Filtering on header](https://github.com/mui/mui-x/issues/6247).
  - [Use the Data Grid internal components outside the grid](https://github.com/mui/mui-x/issues/2522).

- **Improved look & feel**  
   We’re polishing edges in terms of design and usability.

  - [New column menu](https://github.com/mui/mui-x/issues/4929).
  - [New column visibility panel](https://github.com/mui/mui-x/issues/5700).

>   - [Column Resize Bar only on hover](https://github.com/mui/mui-x/issues/1623).

Reviewer 3: ```suggestion
  - [Column resize bar only on hover](https://github.com/mui/mui-x/issues/1623).
```

- **Rockstar feature**
  - [Copy and paste from/to multiple cells](https://github.com/mui/mui-x/issues/199).

> ### Date and Time Pickers

Reviewer 3: Are we doing anything in terms of customizability? Like, expose a hook-only version or a headless version so we can also expose Joy UI version later. I mean, we could have these unstyled API exposed with `unstable_` so we can continue to iterate on it post v6.0.0 stable.

Author: > Are we doing changes in terms of improving customizability?

Yes! That's a good point I missed. 
But indeed the new date fields were built over hooks.

Reviewer 4: We can communicate for the fields yes
For the pickers / views I would avoid to do it for now.
